secret-manager panic - vault

**Describe the bug**:

The secret-manager Pod restarts because of a Panic in secret-manager. This started happening after I created a few `SecretStores` (all Vault, 1 per-namespace) and 1 `ExternalSecret`.

Here's the panic:
```
I1017 15:38:24.813486       1 controller.go:132] Starting secret manager controller: version (v0.2.0) (d6720b56b6b27b68878d6e0e49515874ce0fb8b3)
W1017 15:38:24.814592       1 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I1017 15:38:24.906212       1 listener.go:44] controller-runtime/metrics "msg"="metrics server is starting to listen"  "addr"=":9321"
I1017 15:38:24.906770       1 controller.go:109] Starting manager
I1017 15:38:24.906895       1 leaderelection.go:242] attempting to acquire leader lease  cluster-services/secret-manager-controller...
I1017 15:38:24.907076       1 internal.go:391] controller-runtime/manager "msg"="starting metrics server"  "path"="/metrics"
I1017 15:39:42.380998       1 leaderelection.go:252] successfully acquired lease cluster-services/secret-manager-controller
I1017 15:39:42.381453       1 controller.go:139] controller "msg"="Starting EventSource" "controller"="externalsecret" "reconcilerGroup"="secret-manager.itscontained.io" "reconcilerKind"="ExternalSecret" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"storeRef":{"name":""}},"status":{}}}
I1017 15:39:42.481835       1 controller.go:139] controller "msg"="Starting EventSource" "controller"="externalsecret" "reconcilerGroup"="secret-manager.itscontained.io" "reconcilerKind"="ExternalSecret" "source"={"Type":{"metadata":{"creationTimestamp":null}}}
I1017 15:39:42.482006       1 controller.go:146] controller "msg"="Starting Controller" "controller"="externalsecret" "reconcilerGroup"="secret-manager.itscontained.io" "reconcilerKind"="ExternalSecret"
I1017 15:39:42.482093       1 controller.go:167] controller "msg"="Starting workers" "controller"="externalsecret" "reconcilerGroup"="secret-manager.itscontained.io" "reconcilerKind"="ExternalSecret" "worker count"=1
E1017 15:39:42.487074       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 250 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x175d340, 0x2996dc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/runtime/runtime.go:48 +0x82
panic(0x175d340, 0x2996dc0)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/itscontained/secret-manager/pkg/internal/vault.(*Vault).setToken(0xc000081180, 0x1d314e0, 0xc000122018, 0x1d316a0, 0xc000392a10, 0x1, 0xc0000cc860)
        /workspace/pkg/internal/vault/vault.go:217 +0x2dd
github.com/itscontained/secret-manager/pkg/internal/vault.New(0x1d314e0, 0xc000122018, 0x1d3da40, 0xc00049c6f0, 0x1d45d80, 0xc00024a090, 0x1d60fe0, 0xc0004ce780, 0xc000309bf0, 0x10, ...)
        /workspace/pkg/internal/vault/vault.go:80 +0x177
github.com/itscontained/secret-manager/pkg/internal/store/base.(*Default).New(0x29e6f48, 0x1d314e0, 0xc000122018, 0x1d3da40, 0xc00049c6f0, 0x1d60fe0, 0xc0004ce780, 0x1d45d80, 0xc00024a090, 0x7f717e1409a8, ...)
        /workspace/pkg/internal/store/base/store.go:42 +0x13e
github.com/itscontained/secret-manager/pkg/controller/externalsecret.(*ExternalSecretReconciler).Reconcile.func1(0x0, 0x0)
        /workspace/pkg/controller/externalsecret/controller.go:90 +0x1ea
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0xc000080f00, 0xc000309bf0, 0x10, 0xc000309bd0, 0xd, 0x1d027e0, 0xc0004ce640, 0x1d027e0, 0xc0004ce640)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/controller/controllerutil/controllerutil.go:228 +0x2b
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x1d314e0, 0xc000122018, 0x1d45d80, 0xc00024a090, 0x1d027e0, 0xc0004ce640, 0xc000080f00, 0x1cfed20, 0xc00016d340, 0x0, ...)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/controller/controllerutil/controllerutil.go:202 +0x1db
github.com/itscontained/secret-manager/pkg/controller/externalsecret.(*ExternalSecretReconciler).Reconcile(0xc0004967e0, 0xc000309bf0, 0x10, 0xc000309bd0, 0xd, 0xc0006b4540, 0xc0006e2bd0, 0xc0001d5688, 0xc0001d5680)
        /workspace/pkg/controller/externalsecret/controller.go:84 +0x3b8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004d2000, 0x17c57a0, 0xc000836e00, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:235 +0x284
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004d2000, 0x203000)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209 +0xae
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0004d2000)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0002b9000)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0002b9000, 0x1cf6b40, 0xc0006b4510, 0x1, 0xc000100660)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0002b9000, 0x3b9aca00, 0x0, 0xc000455b01, 0xc000100660)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc0002b9000, 0x3b9aca00, 0xc000100660)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:170 +0x411
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13d535d]

goroutine 250 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/runtime/runtime.go:55 +0x105
panic(0x175d340, 0x2996dc0)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/itscontained/secret-manager/pkg/internal/vault.(*Vault).setToken(0xc000081180, 0x1d314e0, 0xc000122018, 0x1d316a0, 0xc000392a10, 0x1, 0xc0000cc860)
        /workspace/pkg/internal/vault/vault.go:217 +0x2dd
github.com/itscontained/secret-manager/pkg/internal/vault.New(0x1d314e0, 0xc000122018, 0x1d3da40, 0xc00049c6f0, 0x1d45d80, 0xc00024a090, 0x1d60fe0, 0xc0004ce780, 0xc000309bf0, 0x10, ...)
        /workspace/pkg/internal/vault/vault.go:80 +0x177
github.com/itscontained/secret-manager/pkg/internal/store/base.(*Default).New(0x29e6f48, 0x1d314e0, 0xc000122018, 0x1d3da40, 0xc00049c6f0, 0x1d60fe0, 0xc0004ce780, 0x1d45d80, 0xc00024a090, 0x7f717e1409a8, ...)
        /workspace/pkg/internal/store/base/store.go:42 +0x13e
github.com/itscontained/secret-manager/pkg/controller/externalsecret.(*ExternalSecretReconciler).Reconcile.func1(0x0, 0x0)
        /workspace/pkg/controller/externalsecret/controller.go:90 +0x1ea
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0xc000080f00, 0xc000309bf0, 0x10, 0xc000309bd0, 0xd, 0x1d027e0, 0xc0004ce640, 0x1d027e0, 0xc0004ce640)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/controller/controllerutil/controllerutil.go:228 +0x2b
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x1d314e0, 0xc000122018, 0x1d45d80, 0xc00024a090, 0x1d027e0, 0xc0004ce640, 0xc000080f00, 0x1cfed20, 0xc00016d340, 0x0, ...)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/controller/controllerutil/controllerutil.go:202 +0x1db
github.com/itscontained/secret-manager/pkg/controller/externalsecret.(*ExternalSecretReconciler).Reconcile(0xc0004967e0, 0xc000309bf0, 0x10, 0xc000309bd0, 0xd, 0xc0006b4540, 0xc0006e2bd0, 0xc0001d5688, 0xc0001d5680)
        /workspace/pkg/controller/externalsecret/controller.go:84 +0x3b8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004d2000, 0x17c57a0, 0xc000836e00, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:235 +0x284
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004d2000, 0x203000)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209 +0xae
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0004d2000)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0002b9000)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0002b9000, 0x1cf6b40, 0xc0006b4510, 0x1, 0xc000100660)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0002b9000, 0x3b9aca00, 0x0, 0xc000455b01, 0xc000100660)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc0002b9000, 0x3b9aca00, 0xc000100660)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:170 +0x411
```

**Expected behavior**

An error message, not a panic.

**Steps to reproduce the bug**:

* Install secret-manager
* Create a/multiple Vault SecretStore (note the empty "server", though I do not know if that's what's triggering the bug)

```
apiVersion: secret-manager.itscontained.io/v1alpha1
kind: SecretStore
metadata:
  name: vault
  namespace: cluster-services
spec:
  vault:
    auth:
      kubernetes:
        mountPath: cluster-alpha
        role: vault-authentication
    path: secret/data
    server: ""
```

* Create one external secret

```
apiVersion: secret-manager.itscontained.io/v1alpha1
kind: ExternalSecret
metadata:
  name: hello-service
  namespace: cluster-services
spec:
  data:
  - remoteRef:
      name: kv_test/hugues
      property: a
    secretKey: password
  storeRef:
    name: vault
```

**Anything else we need to know?**:

It's possible the authentication on my end is not correctly configured and/or that the paths specified in my ExternalSecret are wrong and don't exist in my Vault.

At any rate, secret-manager shouldn't panic.

**Environment details:**:
- Kubernetes version (e.g. v1.18.9): EKS.3  K8s 1.18
- Secret-Manager version (e.g. v0.2.0): secret-manager 0.2.0, chart 0.1.2
- SecretStore type(e.g. Vault vs AWS SecretManager): Vault
- Install method (e.g. helm or static manifests): Helm

/kind bug

--- a/pkg/internal/vault/vault.go
+++ b/pkg/internal/vault/vault.go
@@ -217,7 +217,7 @@ func (v *Vault) setToken(ctx context.Context, client Client) error {
 	if kubernetesAuth != nil {
 		token, err := v.requestTokenWithKubernetesAuth(ctx, client, kubernetesAuth)
 		if err != nil {
-			return fmt.Errorf("error reading Kubernetes service account token from %s: %s", kubernetesAuth.SecretRef.Name, err.Error())
+			return fmt.Errorf("error reading Kubernetes service account token. error: %s", err.Error())
 		}
 		client.SetToken(token)
 		return nil

